### DISCUSSION TITLE
Conditionally open style manager properties depending on the block type

### DISCUSSION THREAD

**Discussion by saharhagekr** [GrapesJS/grapesjs#5490] [Q&A] [answered]
Hello!
Thanks for providing this amazing  library!

I have a question that i don't know if feasible or not, and couldn't find a similar thread/discussion.

Is there a way to open the style manager depending on the block type? Only show properties related to this block type.
Like for example for the texts only have the text editor properties available. For the images, have the selected image in the side banner, with a button to change the image, and with image properties underneath the image.

If this is feasible, can you please guide me on where to find some similar implementation so i can understand how to do it?

Thank you!!

**Comment by mosh-tudor** [comment, marked as answer]
You can filter unneeded styles using `stylable`:
https://grapesjs.com/docs/api/component.html#component

> `stylable` **([Boolean][3] | [Array][5]<[String][1]>)?** True if it's possible to style the component.
>     You can also indicate an array of CSS properties which is possible to style, eg. `['color', 'width']`, all other properties
>     will be hidden from the style manager. Default: `true`
